pa4/checker-pa4.ml

```diff
@@ -768,26 +778,36 @@ let main () = begin
 	(* Consolidated List of User and Base classes *)
     (* This adds the bases classes to our class inheritence mapping so that they aren't treated as
 -	undeclared classes. *)
-        let base_classes = ["Bool"; "IO"; "Int"; "String"; "Object"] in
-        let user_classes = List.map (fun ((_,cname),_,_) -> cname) ast in
+    let base_classes = ["Bool"; "IO"; "Int"; "String"; "Object"] in
+    let user_classes = List.map (fun ((_,cname),_,_) -> cname) ast in
 	let all_classes = base_classes @ user_classes in
 	let all_classes = List.sort compare all_classes in
-        let vertices = ref [] in
+    let vertices = ref [] in
 
 	(* Check of Main class, else return *)
 	if not (List.mem "Main" user_classes) then begin
 		printf "ERROR: 0: Type-Check: class Main not found\n";
 		exit 1
 	end;
 
-	(* populate parent map *)
+	(* Set up global maps *)
+	let m_e = empty_method_env () in (* add features to obj env *)
```

Comment:
Is there a reason that this variable is not defined in another location? I would think that this would only be relevant when we are checking a specific class, so it could go below line 795.